model/task/generate_tasks_estimations.go

```diff
@@ -0,0 +1,72 @@
+package task
+
+import (
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/utility"
+	adb "github.com/mongodb/anser/db"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	numRevisionsToSearch = 10
+)
+
+func (t *Task) setGenerateTasksEstimations() error {
+	if !t.GenerateTask || (t.EstimatedNumGeneratedTasks != nil && t.EstimatedNumActivatedGeneratedTasks != nil) {
+		return nil
+	}
+
+	query := db.Query(
+		ByPreviousCommit(
+			t.BuildVariant,
+			t.DisplayName,
+			t.Project,
+			evergreen.RepotrackerVersionRequester,
+			t.RevisionOrderNumber,
+		),
+	).Sort([]string{"-" + RevisionOrderNumberKey}).Limit(numRevisionsToSearch)
+
+	tasks := []Task{}
+	err := db.FindAllQ(Collection, query, &tasks)
+	if adb.ResultsNotFound(err) {
+		return errors.Errorf("Tasks '%s' in '%s' not found", t.DisplayName, t.Project)
```

Comment:
Do we want to use the task ID here instead? Seems like the task ID would be more helpful to determine which particular task had the issue.

Same with the other returns.